Lib/test/test_ast.py

```diff
@@ -399,6 +399,9 @@ def test_isinstance(self):
         self.assertFalse(isinstance(ast.Constant(), ast.NameConstant))
         self.assertFalse(isinstance(ast.Constant(), ast.Ellipsis))
 
+        class S(str): pass
+        self.assertTrue(isinstance(ast.Constant(S('42')), ast.Str))
```

Comment:
Can you please add this test?
self.assertFalse(isinstance(ast.Constant(S('42')), ast.Num))

Reply:
(maybe instanciate the ast object only once)

Reply:
I think it looks clearer if the ast object is instantiated at the same line. In most tests different ast objects are tested in sequential lines.